compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Markdown.hs

```diff
@@ -9,98 +9,124 @@ module DA.Daml.Doc.Render.Markdown
 
 import DA.Daml.Doc.Types
 import DA.Daml.Doc.Render.Util
+import DA.Daml.Doc.Render.Monoid
 
 import           Data.Maybe
 import qualified Data.Text as T
-import           Data.List (intersperse)
+import Data.List.Extra
 
-renderSimpleMD :: ModuleDoc -> T.Text
+
+-- | Declare anchor and generate HTML tag to insert in renderer output.
+withAnchorTag :: Maybe Anchor -> (T.Text -> RenderOut) -> RenderOut
+withAnchorTag Nothing fn = fn ""
+withAnchorTag (Just anchor) fn = mconcat
+    [ renderDeclareAnchor anchor
+    , fn $ T.concat ["<a name=\"", unAnchor anchor, "\"></a>"]
+    ]
+
+-- | Declare anchor and output HTML tag between two pieces of text.
+renderAnchorInfix :: T.Text -> Maybe Anchor ->  T.Text -> RenderOut
+renderAnchorInfix pre anchor post =
+    withAnchorTag anchor (\tag -> renderLine $ T.concat [ pre, tag, post ])
+
+-- | Render doc text. If Nothing, renders an empty line. If Just, renders
+-- doc text block between two empty lines.
```

Comment:
That explanation sounds like we're rendering a paragraph. Is that correct? If so, we should reflect it in the name.